cli/command/image/inspect_test.go

```diff
@@ -0,0 +1,92 @@
+package image
+
+import (
+"bytes"
+"fmt"
+"github.com/docker/docker/cli/internal/test"
+"github.com/docker/docker/pkg/testutil/assert"
+"github.com/docker/docker/pkg/testutil/golden"
+"io/ioutil"
+"testing"
+	"github.com/docker/docker/api/types"
+)
+
+func TestNewInspectCommandErrors(t *testing.T) {
+	testCases := []struct {
+		name            string
+		args            []string
+	}{
+		{
+			name: "wrong-args",
+			args: []string{},
+		},
+	}
+	for _, tc := range testCases {
+		buf := new(bytes.Buffer)
+		cmd := newInspectCommand(test.NewFakeCli(&fakeClient{}, buf))
+		cmd.SetOutput(ioutil.Discard)
+		cmd.SetArgs(tc.args)
+		err := cmd.Execute()
+		assert.NotNil(t, err)
+		actual := err.Error()
+		expected := string(golden.Get(t, []byte(actual), fmt.Sprintf("inspect-command-errors.%s.golden", tc.name))[:])
+		assert.EqualNormalizedString(t, assert.RemoveSpace, actual, expected)
+	}
+}
+
+func TestNewInspectCommandSuccess(t *testing.T) {
+	imageInspectFuncCount := 0
+	testCases := []struct {
+		name             string
+		args             []string
+		imageCount       int
+		imageInspectFunc func(image string) (types.ImageInspect, []byte, error)
+	}{
+		{
+			name: "simple",
+			args: []string{"image"},
+			imageCount: 1,
+			imageInspectFunc: func(image string) (types.ImageInspect, []byte, error) {
+				imageInspectFuncCount++
+				assert.Equal(t, image, "image")
+				return types.ImageInspect{}, nil, nil
+			},
+		},
+		{
+			name: "format",
+			imageCount: 1,
+			args: []string{"--format='{{.ID}}'", "image"},
+			imageInspectFunc: func(image string) (types.ImageInspect, []byte, error) {
+				imageInspectFuncCount++
+				return types.ImageInspect{ID: image}, nil, nil
+			},
+		},
+		{
+			name: "simple-many",
+			args: []string{"image1", "image2"},
+			imageCount: 2,
+			imageInspectFunc: func(image string) (types.ImageInspect, []byte, error) {
+				imageInspectFuncCount++
+				if imageInspectFuncCount == 1 {
+					assert.Equal(t, image, "image1")
+				} else {
+					assert.Equal(t, image, "image2")
+				}
+				return types.ImageInspect{}, nil, nil
+			},
+		},
+	}
+	for _, tc := range testCases {
+		imageInspectFuncCount = 0
```

Comment:
I want to validate `cli.Client().ImageInspect()` is being called X amount of times. Maybe there's a more readable way to do it.